migration-script/get_addresses.py

```diff
@@ -0,0 +1,269 @@
+    signed_tx = w3.eth.account.signTransaction(tx, private_key=PRIVATE_KEY)
+    w3.eth.sendRawTransaction(signed_tx.rawTransaction)
+    print(f"Transaction hash: {signed_tx.hash.hex()}")
+    print(f"Waiting for receipt")
+    w3.eth.wait_for_transaction_receipt(signed_tx.hash, timeout=120)
+
+def main():
+    w3 = Web3(Web3.HTTPProvider(DEFAULT_RPC))
+    first_method = True
+    addresses = []
+    if first_method:
+        request = f"https://api.arbiscan.io/api?module=account&action=txlist&address={LENT_MYC_ADDRESS}&startblock=0&endblock=19345791&sort=asc&apikey={ETHERSCAN_API_KEY}"
+        response = requests.get(request)
+        print(len(response.json()['result']))
+        print("hi")
+        time.sleep(5)
+        for tx in response.json()["result"]:
+            # if w3.toChecksumAddress(tx["from"]) not in addresses:
+                addresses.append(w3.toChecksumAddress(tx["from"]))
+        request = f"https://api.arbiscan.io/api?module=account&action=txlist&address={LENT_MYC_ADDRESS}&startblock=19345791&endblock=29018687&sort=asc&apikey={ETHERSCAN_API_KEY}"
+        response = requests.get(request)
+        print(len(response.json()['result']))
+        print("hi")
+        time.sleep(5)
+        for tx in response.json()["result"]:
+            # if w3.toChecksumAddress(tx["from"]) not in addresses:
+                addresses.append(w3.toChecksumAddress(tx["from"]))
+
+        request = f"https://api.arbiscan.io/api?module=account&action=txlist&address={LENT_MYC_ADDRESS}&startblock=29018687&endblock=38691583&sort=asc&apikey={ETHERSCAN_API_KEY}"
+        response = requests.get(request)
+        print(len(response.json()['result']))
+        print("hi")
+        time.sleep(5)
+        for tx in response.json()["result"]:
+            # if w3.toChecksumAddress(tx["from"]) not in addresses:
+                addresses.append(w3.toChecksumAddress(tx["from"]))
+
+        request = f"https://api.arbiscan.io/api?module=account&action=txlist&address={LENT_MYC_ADDRESS}&startblock=38691583&endblock=999999999&sort=asc&apikey={ETHERSCAN_API_KEY}"
+        response = requests.get(request)
+        print(len(response.json()['result']))
+        print("hi")
+        time.sleep(5)
+        for tx in response.json()["result"]:
+            # if w3.toChecksumAddress(tx["from"]) not in addresses:
+                addresses.append(w3.toChecksumAddress(tx["from"]))
+    else:
+
+        request  = f"https://api.etherscan.io/api?module=token&action=tokenholderlist&contractaddress={LENT_MYC_ADDRESS}&page=1&offset=10&apikey={ETHERSCAN_API_KEY}"
+        response = requests.get(request)
+
+        headers = {
+            'Authorization': 'Basic Y2tleV9mNjY5MzViYWNhZTA0MmU3ODgxMzAwMjk4MGI6',
+        }
+
+        response = requests.get('https://api.covalenthq.com/v1/42161/tokens/0x9B225FF56C48671d4D04786De068Ed8b88b672d6/token_holders/?quote-currency=USD&format=JSON&page-number=0&block-height=latest&page-size=100&key=ckey_7a1ed71b2b294e33bd1bd49e7c4')
```

Comment:
Why the call the etherscan here if you're just gonna overwrite `response` a few lines later?

Reply:
Yeah this was all part of that last minute fix. Will neaten up